workers/price-reporter/src/manager.rs

```diff
@@ -97,29 +98,127 @@ impl AtomicPriceStreamState {
     }
 }
 
-/// A shareable mapping between exchange and the most recent state of their
-/// price stream for a given pair
+/// A shareable mapping between (exchange, base, quote) and the most recent
+/// state of the associated price stream
 #[derive(Clone, Debug)]
-pub struct ExchangeStates(HashMap<Exchange, Arc<AtomicPriceStreamState>>);
-
-impl ExchangeStates {
-    /// Construct a new exchange states instance from a set of exchanges
-    pub fn new_from_exchanges(exchanges: &[Exchange]) -> Self {
-        let mut exchange_states = ExchangeStates(HashMap::new());
-        for exchange in exchanges {
-            exchange_states.0.insert(*exchange, Arc::new(AtomicPriceStreamState::default()));
-        }
-        exchange_states
+pub struct SharedPriceStreamStates(
```

Comment:
See my comment below about the connection muxer, I don't think it should handle price conversion. Another reason is that we pay for it here. We need a more complex structure under the assumption that the connection muxer should be handling possibly multiple token pairs. 

And that is only to facilitate _two_ pairs (the base pair and the conversion pair); seems like far to heavy a change to handle two pairs

Reply:
I would much rather use the `SharedPriceStreamStates` in the `Reporter` than not, it centralizes the logic used to aggregate prices (including price conversion, deviation checks, etc.) between the native and external price reporter. I'll explore a sensible way to avoid passing the struct down to the `ConnectionMuxer`, but I think overall, it largely simplifies the price reporting code.